libnavigation-core/src/main/java/com/mapbox/navigation/core/directions/session/RoutesRequestCallback.kt

```diff
@@ -0,0 +1,40 @@
+package com.mapbox.navigation.core.directions.session
+
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.navigation.base.route.Router
+
+/**
+ * Interface definition for a callback associated with a routes request.
+ */
+interface RoutesRequestCallback {
+
+    /**
+     * Invoked whenever a new set of routes is available.
+     *
+     * The return type is a list that's going to be assigned to and maintained by the [DirectionsSession].
+     * The route at index 0, if exist, will be treated as the primary route for 'Active Guidance' and location map-matching.
+     *
+     * If an empty list is returned, this request will effectively be ignored by the [DirectionsSession].
+     *
+     * @param routes list of routes returned by a [Router]
+     * @return a list of routes that will be assigned to the session where route at index 0 is the primary one
+     */
```

Comment:
Similarly, if my understanding is correct then events like `faster route`, `re route` and `route refresh` won't result in invocation of these methods. It will be helpful to add that here.

Reply:
This is TBD. We could cache this initial request callback and reuse it whenever we detect a faster route, re-route or route refresh. We might also introduce new ones. This will all clear up when we move forward with those impls.